ui/components/app/signature-request/signature-request.js

```diff
@@ -0,0 +1,355 @@
+  const [messageIsScrollable, setMessageIsScrollable] = useState(false);
+
+  const {
+    id,
+    type,
+    msgParams: { from, data, origin, version },
+  } = txData;
+
+  // not forwarded to component
+  const hardwareWalletRequiresConnection = useSelector((state) =>
+    doesAddressRequireLedgerHidConnection(state, from),
+  );
+  const { chainId, rpcPrefs } = useSelector(getProviderConfig);
+  const unapprovedMessagesCount = useSelector(getTotalUnapprovedMessagesCount);
+  const subjectMetadata = useSelector(getSubjectMetadata);
+  const isLedgerWallet = useSelector((state) => isAddressLedger(state, from));
+  const { handleCancelAll } = useRejectTransactionModal();
+
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  // Used to show a warning if the signing account is not the selected account
+  // Largely relevant for contract wallet custodians
+  const selectedAccount = useSelector(getSelectedAccount);
+  const mmiActions = mmiActionsFactory();
+  const accountType = useSelector(getAccountType);
+  const isNotification = getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION;
+  const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
+  const { address } = getAccountByAddress(allAccounts, from) || {};
+  const { custodySignFn } = useMMICustodySignMessage();
+  ///: END:ONLY_INCLUDE_IN
+
+  useEffect(() => {
+    setMessageIsScrollable(
+      messageRootRef?.scrollHeight > messageRootRef?.clientHeight,
+    );
+  }, [messageRootRef]);
+
+  const targetSubjectMetadata = txData.msgParams.origin
+    ? subjectMetadata?.[txData.msgParams.origin]
+    : null;
+
+  const parseMessage = () => {
+    const { message, domain = {}, primaryType, types } = JSON.parse(data);
+    const sanitizedMessage = sanitizeMessage(message, primaryType, types);
+    return { sanitizedMessage, domain, primaryType };
+  };
+
+  const onSign = async () => {
+    await dispatch(resolvePendingApproval(id));
+    completedTx(id);
+    ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+    if (accountType === 'custody') {
+      await custodySignFn(txData);
+    }
+    ///: END:ONLY_INCLUDE_IN
+
+    trackEvent({
+      category: MetaMetricsEventCategory.Transactions,
+      event: 'Confirm',
+      properties: {
+        action: 'Sign Request',
+        legacy_event: true,
+        type,
+        version,
+      },
+    });
+  };
+
+  const onCancel = async () => {
+    await dispatch(
+      rejectPendingApproval(
+        id,
+        serializeError(ethErrors.provider.userRejectedRequest()),
+      ),
+    );
+    trackEvent({
+      category: MetaMetricsEventCategory.Transactions,
+      event: 'Cancel',
+      properties: {
+        action: 'Sign Request',
+        legacy_event: true,
+        type,
+        version,
+      },
+    });
+  };
+
+  const {
+    sanitizedMessage,
+    domain: { verifyingContract },
+    primaryType,
+  } = parseMessage();
+
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  useEffect(() => {
+    if (txData.custodyId) {
+      dispatch(
```

Comment:
For some reason the `showCustodianDeepLink` was not wrapped in a dispatch on the `mmiMapDispatchToProps`. Not really sure why but might be worth keeping the same logic (or ask someone from MMI, just to be sure that not using dispacth was intended).

Reply:
Talked with @zone-live about this. Let's keep it consistent to the previous implementation and remove the `dispatch` wrapping it.